### DISCUSSION TITLE
Add a text editor for the File Library

### DISCUSSION THREAD

**Discussion by benhaynes** [directus/directus#3142] [Feature Requests]
This would allow users to create, view, and edit raw files using a text/code editor.

**Comment by Thermal-Shock** [comment]
This would be very useful to me right now. Essentially invoking the codemirror interface for any supported file extensions. 

This would make it possible to use the file library like a file system of sorts instead of having a parallel git repo and vscode. 

If it was possible, I'd even say see if there is a way to embed the vscode interface and use the file library as a remote file system, but at this point simply having codemirror edit support for all supported file extensions would be very useful.

**Reply by rijkvanzanten** [reply]
> I'd even say see if there is a way to embed the vscode interface and use the file library as a remote file system

👀 that's a very interesting idea. Could be a great custom module extension as well

**Comment by fitu86** [comment]
I have been trying to use the code type field, the idea is through flows instead of storing the file, it only saves the content in the field, If you need to download the file, a manual flow is started from the item to generate the file, I haven't succeeded yet, any recommendations?

**Comment by jaads** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
